corehq/apps/cachehq/cachemodels.py

```diff
@@ -17,17 +17,11 @@ class UserGenerationCache(GenerationCache):
     generation_key = '#gen#couch_user#'
     doc_types = ['CommCareUser', 'CouchUser', 'WebUser']
     views = [
-        "users_extra/phones_to_domains",
-        "users_extra/users_over_time",
-        "users_extra/emails",
         "users/by_domain",
         "users/phone_users_by_domain",
-        "users/web_users_by_domain",
         "users/by_default_phone",
-        "users/admins_by_domain",
```

Comment:
I think removing these (well actually, removing the associated `map.js` and `reduce.js` files) but not removing all other "users" views (i.e., not removing the entire ` corehq/apps/users/_design` directory) means Couch will rebuild all other user views. Do we want that to happen?

This understanding of how Couch works is based on what @dannyroberts told me a few days ago. Maybe I misunderstood?

Reply:
Yeah, my understanding is that both `users` and `users_extra` will be reindexed, and Danny is the person who best knows how burdensome that is.

Reply:
If we do remove these views, we might consider moving the remaining views each into their own design doc (in `corehq/couchapps`), assuming that's not much more effort.  That way we can subsequently delete say `domain/old_users` without needing to rebuild the rest.

Reply:
Hi @dannyroberts , any thoughts on this?

Reply:
Yeah I think it would be quite burdensome if it were in the main db, but I'm guessing this is only in the commcarehq__users db, right? That has about 160k documents, which not nothing but also not out of the question. Enough that we might want to do it on a weekend unless we had a better way to estimate it.

I'd also ask that we hold off on merging this until a week or so after we've upgraded to couch 3 just to space out any small amount of risk there might be.

Reply:
Yeah, this should just be commcarehq__users. Happy to hold off until couch 3 has been live for a bit.

Reply:
I believe Couch 3 has been live for a while now, any objections to merging it this Friday?

@dannyroberts @skodde @calellowitz

Reply:
No objections here!

Reply:
@orangejenny @dannyroberts should we hold off a little bit longer until I remove the `roles_by_domain` view (hopefully in the next 2 weeks)?

Reply:
Sure, works for me.

Reply:
https://github.com/dimagi/commcare-hq/pull/30179

Reply:
@snopoke Is https://github.com/dimagi/commcare-hq/pull/30179 (and https://github.com/dimagi/commcare-hq/pull/30178) ready to merge today?

Reply:
#30178 can be merged but I was going to wait before removing the couch view otherwise rollback required rebuilding the view

Reply:
okay, so should I hold off on merging this PR and preindexing until next week? (there's no hurry)

Reply:
yes, thanks. I've merged the other PR so that will go out in the next deploy. Once that has been live for a week we can merge these.

Reply:
cool, sounds good

Reply:
postponing a week because the Thursday morning deploy didn't go through, so https://github.com/dimagi/commcare-hq/pull/30178 isn't live yet

Reply:
actually postponing a few weeks because I'm off the next 2 Fridays. @snopoke if you want to merge this and preindex next Friday go for it, and if you don't get to it, I will eventually.